server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -223,7 +222,41 @@ private void run() {
                 continue;
             }
 
-            ActionListener<AcknowledgedResponse> listener = ActionListener.wrap(r -> countDown.onSuccess(), onError);
+            var clusterStateListener = new FutureActionListener<>(AcknowledgedResponse::isAcknowledged);
+            clusterStateListener
+                .whenComplete(
+                    (r, e) -> {
+                        // Only check for new relations and partitions if deleted relations and partitions are processed
+                        var stateFuture = getPublicationsState(subscription, client);
```

Comment:
Is this intentionally not composing? `clusterStateListener.whenComplete` will complete without waiting for the publicationsState future

Reply:
no, thx for catching!